pkg/generators/factory.go

```diff
@@ -0,0 +1,40 @@
+package generators
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+)
+
+//Generator is a template engine
+type Generator struct {
+	*template.Template
+}
+
+//New creates an instance of a template engine for a set of templates
+func New(name string, addFunctions *template.FuncMap, templates ...string) (*Generator, error) {
+	allFunctions := funcMap
```

Comment:
should you use a deep copy here?  `allFunctions := reflect.DeepCopy(funcMap)`?  otherwise, you will be editing the shared `funcMap`

Reply:
added copy loop